app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,64 @@
+class AlbumsController < ApplicationController
+  def get_album
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def index
+    @albums = Album.all
+    @albums_sort = @albums.sort_by{ |album| album[:rank] }.reverse
+  end
+
+  def show
+    get_album
+  end
+
+  def new
+    @album = Album.new
+    @title = "Add New album"
+    @action = "create"
+  end
+
+  def create
+    @album = Album.create(album_params[:album])
+    @album.update(:rank => 0)
+    if @album.save
+      redirect_to "/albums"
+    else
+      render "/albums/new"
```

Comment:
You can refer to actions in the same controller with symbol notation and with built-in rails path helpers as well. Eg, `redirect_to :index`, `redirect_to albums_path`, `render :new`, `render new_album`. To see a list of path helpers available to you, use the command `rake routes` in your terminal. Add `_path` or `_url` to any prefix to get the matching path helper.